projects/angular/src/forms/input/input.spec.ts

```diff
@@ -39,5 +42,48 @@ export default function (): void {
     ControlStandaloneSpec(StandaloneUseTest);
     TemplateDrivenSpec(ClrInputContainer, ClrInput, TemplateDrivenTest, 'clr-input');
     ReactiveSpec(ClrInputContainer, ClrInput, ReactiveTest, 'clr-input');
+    inputSpec('tempate-driven', ClrInputContainer, ClrInput, TemplateDrivenTest);
+    inputSpec('reactive', ClrInputContainer, ClrInput, ReactiveTest);
+  });
+}
+
+function inputSpec(description, testContainer, testControl, testComponent) {
+  describe('Input specific value change tests ' + description, () => {
+    let control, fixture;
+
+    beforeEach(() => {
+      TestBed.configureTestingModule({
+        imports: [FormsModule, ClrCommonFormsModule, ReactiveFormsModule],
+        declarations: [testContainer, testControl, testComponent],
+      });
+      fixture = TestBed.createComponent(testComponent);
+      control = fixture.debugElement.query(By.directive(testControl));
+      fixture.detectChanges();
+    });
+
+    it('should handle valueChanges calls', () => {
+      // control must be both invalid and blurred to register the validity
+      let valueChanges = 0;
+      control.injector.get(NgControl).control.valueChanges.subscribe(() => {
```

Comment:
Isn't this test essentially testing Angular code, not Clarity-specific code? I don't think we should have tests that essentially re-test Angular.

Reply:
The 'valueChanges' are either triggered by the Angular Framework itself, or forcibly by us in the onBlur handler.
I want to test the effect on this observable, not the implementation detail from our side, as we used to do.
Essentially, the bug was that this handler gets called one extra time. Technically, the test is useless, because the failure can only be reproduced when Clarity is consumed as library and imported before the FormsModule.
For me, a unit test serves 2 purposes - provide testing automation and support the maintenance by defining use case scenarios.
Usually they serve both in the same time, but I added this test only with the second purpose in mind - to define the expectation and help understanding the use case if future problems arise in this area.